### PR TITLE
Feature/wopi editing

### PR DESCRIPTION
**Pull request by VolshebnikQ**

### CODE REVIEW

File: web/documentserver-example/nodejs/helpers/docManager.js [hunk 1]
```diff
@@ -102,12 +102,46 @@ docManager.getCorrectName = function (fileName, userAddress) {
...
+
 // create demo document
-docManager.createDemo = function (isSample, fileExt, userid, username) {
+docManager.createDemo = function (isSample, fileExt, userid, username, wopi) {
+
     const demoName = (isSample ? "sample" : "new") + "." + fileExt;
     const fileName = docManager.getCorrectName(demoName);  // get the correct file name if such a name already exists
 
-    docManager.copyFile(path.join(__dirname, "..","public", "assets", isSample ? "sample" : "new", demoName), docManager.storagePath(fileName));   // copy sample document of a necessary extension to the storage path
+    if (wopi){
```
Comment (LinneyS) [inline web/documentserver-example/nodejs/helpers/docManager.js:140]: not used

File: web/documentserver-example/nodejs/helpers/docManager.js [hunk 2]
```diff
@@ -102,12 +102,46 @@ docManager.getCorrectName = function (fileName, userAddress) {
     return name;
 };
 
+// processes a request editnew
+docManager.RequestEditnew = function (req, fileName, user) {
+    if (req.params['id'] != fileName){  // processes a repeated request editnew
+        docManager.fileRemove(req.params['id'], req.host);
+        fileName = docManager.getCorrectName(req.params['id']);
+
+        docManager.fileSizeZero(fileName);
```
Comment (LinneyS) [inline web/documentserver-example/nodejs/helpers/docManager.js:111]: duplicate

File: web/documentserver-example/nodejs/helpers/docManager.js [hunk 3]
```diff
@@ -102,12 +102,46 @@ docManager.getCorrectName = function (fileName, userAddress) {
...
+    docManager.saveFileData(fileName, user.id, user.name);
+
+    return fileName;
+}
+
+// delete a file with its history
+docManager.fileRemove = function (fileName) {
+    var path = docManager.storagePath(fileName);
+    docManager.cleanFolderRecursive(path+"-history\\",true)
+    fileSystem.unlinkSync(path);
```
Comment (LinneyS) [inline web/documentserver-example/nodejs/helpers/docManager.js:124]: use in app.js 
`app.delete("/file"`

**Review by LinneyS — CHANGES_REQUESTED** [review]

**Review by Eloncase — APPROVED** [review]

**Review by LinneyS — CHANGES_REQUESTED** [review]
Delete creation with sample data

Fix `version` in file info

**Pull request merged**